modules/trusted-firmware-m/Kconfig

```
@@ -16,7 +16,22 @@
 if BUILD_WITH_TFM
 
 config TFM_IPC
-	default y if TFM_PROFILE_TYPE_MINIMAL
+	default y
+
+config TFM_ISOLATION_LEVEL
+	int "Isolation level setting." if (TFM_PROFILE_TYPE_NOT_SET && TFM_IPC)
+	range 1 3
+	depends on BUILD_WITH_TFM
+	default 1 if TFM_PROFILE_TYPE_SMALL || !TFM_IPC
+	default 3 if TFM_PROFILE_TYPE_LARGE
+	default 2
+	help
+	  Manually set the required TFM isolation level. Possible values are
+	  1,2 or 3; the default is set by build configuration. When TF-M
+	  Profile option is supplied, do not allow manual setting of the
+	  isolation level, as it is determined by the profile setting.
+	  As isolation levels 2 and 3 require PSA_API (TFM_IPC) support,
+	  force level 1 when TFM_IPC is not enabled.
```

I think this is enough to override the default value. We don't have to fully redeclare it.
```suggestion
config TFM_ISOLATION_LEVEL
	int
	default 2
```

---

Ok, I tried it and you are right. I was not sure if we will be able to configure it this way but it seems that we can. So I changed this!